sql/modules/Roles.sql

```diff
@@ -356,18 +421,75 @@ SELECT lsmb__grant_perms('contact_read', 'file_entity', 'SELECT');
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_contact',
+                         $DOC$
+                         This role allows access to contact data (e-mail, phone, etc) of all kinds of contacts (customer/vendor/...).
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_referral',
+                         $DOC$
+                         This role allows access to referral contact data.
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_lead',
+                         $DOC$
+                         This role allows access to sales lead contact data.
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_hot_lead',
+                         $DOC$
+                         This role allows access to hot sales lead contact data.
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_cold_lead',
+                         $DOC$
+                         This role allows access to cold sales lead contact data.
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_sub_contractor',
+                         $DOC$
+                         This role allows access to subcontractor contact data.
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+SELECT lsmb__create_role('contact_class_robot',
+                         $DOC$
+                         This role allows access to robot (automated process, acting on behalf of...) contact data.
+                         Combine with 'contact_read', 'contact_create', 'contact_edit' to determine the type of access granted.
+                         $DOC$
+);
+
+SELECT lsmb__create_role('contact_create',
+                         $DOC$
+                         This role allows creation of new contacts.
+
+                         Combine this role with one or more 'contact_class_*' roles to be able to access
+                         contacts of the specific class. By itself, this role does not provide sufficient rights.
+                         $DOC$
+);
```

Comment:
I'd suggest a similar wording as suggested for `contact_read`.

Reply:
Done.